libminifi/include/SchedulingAgent.h

```diff
@@ -131,11 +136,16 @@ class SchedulingAgent {
     auto pool = utils::ThreadPool<uint64_t>(csThreads, false, controller_service_provider, "SchedulingAgent");
     thread_pool_ = std::move(pool);
     thread_pool_.start();
+
+    std::function<void(void)> f = std::bind(&SchedulingAgent::watchDogFunc, this);
+    watchDogTimer_.reset(new utils::CallBackTimer(std::chrono::milliseconds(SCHEDULING_WATCHDOG_CHECK_PERIOD), f));
+    watchDogTimer_->start();
   }
   // Destructor
   virtual ~SchedulingAgent() {
-
+    watchDogTimer_->stop();
```

Comment:
Removed the whole  dtor.

Reply:
By removing the dtor, you've made it non-virtual, which is also problematic because there are derived classes from `SchedulingAgent`.
I'd suggest `virtual ~SchedulingAgent() = default;`, possibly `noexcept` if all the member and derived class dtors are also `noexcept`.

Reply:
I had to put the stop call back to the dtor, added a comment there to make sure.